### PR TITLE
Ignore Bootstrap Token secrets that don't use predictable names.

### PR DESCRIPTION
**Pull request by jbeda**
This aligns with spec changes coming in https://github.com/kubernetes/community/pull/381.

### CODE REVIEW

File: pkg/controller/bootstrap/util.go [hunk 1]
```diff
@@ -36,13 +40,32 @@ func getSecretString(secret *v1.Secret, key string) string {
...
+	}
+
 	tokenID = getSecretString(secret, bootstrapapi.BootstrapTokenIDKey)
 	if len(tokenID) == 0 {
 		glog.V(3).Infof("No %s key in %s/%s Secret", bootstrapapi.BootstrapTokenIDKey, secret.Namespace, secret.Name)
 		return "", "", false
 	}
 
+	if nameTokenID != tokenID {
+		glog.V(3).Infof("Token ID (%s) doesn't match secret name: %s", tokenID, nameTokenID)
```
Comment (luxas) [inline pkg/controller/bootstrap/util.go]: It should stop here, right? This is an invalid Secret, so `return "", "", false`

File: pkg/controller/bootstrap/util_test.go [hunk 2]
```diff
@@ -135,3 +137,48 @@ func TestValidateSecret(t *testing.T) {
...
+			bootstrapapi.BootstrapTokenUsageSigningKey: []byte("true"),
+		},
+	}
+
+	_, _, ok := validateSecretForSigning(secret)
+	if ok {
+		t.Errorf("Token validation should fail with bad name")
+	}
+}
+
```
Comment (luxas) [inline pkg/controller/bootstrap/util_test.go]: Add case when `metadata.name != data["token-id"]`?

File: pkg/controller/bootstrap/util_test.go [hunk 3]
```diff
@@ -135,3 +137,48 @@ func TestValidateSecret(t *testing.T) {
...
+func TestParseSecretName(t *testing.T) {
+	tokenID, ok := parseSecretName("bootstrap-token-abc123")
+	assert.True(t, ok, "parseSecretName should accept valid name")
+	assert.Equal(t, "abc123", tokenID, "parseSecretName should return token ID")
+
+	_, ok = parseSecretName("")
+	assert.False(t, ok, "parseSecretName should reject blank name")
+
+	_, ok = parseSecretName("abc123")
+	assert.False(t, ok, "parseSecretName should no prefix")
```
Comment (luxas) [inline pkg/controller/bootstrap/util_test.go]: parseSecretName has no prefix

**Review by luxas — COMMENTED** [review]
Minor comments only, I hope we can merge this quickly

File: pkg/controller/bootstrap/util.go (hunk 1 above)
Comment (jbeda) [inline pkg/controller/bootstrap/util.go]: Arg -- right.  I should have a test for that too.

**Comment by luxas** [comment]
Also:
```
W0220 10:41:25.358] Errors from golint:
W0220 10:41:25.358] pkg/bootstrap/api/types.go:24:2: comment on exported const BootstrapTokenSecretPrefix should be of the form "BootstrapTokenSecretPrefix ..."
W0220 10:41:25.358] 
W0220 10:41:25.358] Please fix the above errors. You can test via "golint" and commit the result.
```

**Comment by jbeda** [comment]
Didn't know we were enforcing lint.  Cool.

File: pkg/controller/bootstrap/util_test.go (hunk 2 above)
Comment (jbeda) [inline pkg/controller/bootstrap/util_test.go]: added

**Comment by jbeda** [comment]
Ok -- changes submitted.  I'm going to be AFK for a bit at lunch with family.

**Comment by luxas** [comment]
Sure, let's hope tests are passing

/lgtm

**Comment by luxas** [comment]
@k8s-bot gce etcd3 e2e test this

**Pull request merged**